### PR TITLE
fix(plonky2_prove_failure/signed_lt_arr): Fix test types and data

### PR DESCRIPTION
**Pull request by Syndamia**
The `signed_lt_arr` test had some issues, mainly:

1. inside the main function, the type was unsigned, not signed
2. the values in Prover.toml don't represent the ones in comments (iterating through different bit patterns)

This PR fixes those issues.

### CODE REVIEW

File: test_programs/plonky2_prove_failure/signed_lt_arr/Prover.toml [hunk 1]
```diff
@@ -1,15 +1,15 @@
 # all combinations of the low 4 bits, i.e. 0000xxxx in binary
-x1 = [0, -1, -2, -3, -4, -5, -6, -7, -8, -9,-10,-11,-12,-13,-14,-15]
-y1 = [0, -1, -2, -3, -4, -5, -6, -7, -8, -9,-10,-11,-12,-13,-14,-15]
+x1 = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9,10,11,12,13,14,15]
+y1 = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9,10,11,12,13,14,15]
 # all combinations of the high 4 bits, i.e. xxxx0000 in binary
-x2 = [0,-16,-32,-48,-64,-80,-96,-112,-128,-144,-160,-176,-192,-208,-224,-240]
-y2 = [0,-16,-32,-48,-64,-80,-96,-112,-128,-144,-160,-176,-192,-208,-224,-240]
+x2 = [-16,-32,-48,-64,-80,-96,-112,-128,112,96,80,64,48,32,16,0]
```
Comment (nickysn) [inline test_programs/plonky2_prove_failure/signed_lt_arr/Prover.toml:5]: The are the correct numbers, but in the wrong order. These must be in sorted order with regards to signed comparison. The correct order is:
-128,-127,-126,-125,-64,-63,-62,-61,0,1,2,3,64,65,66,67

File: test_programs/plonky2_prove_failure/signed_lt_arr/Prover.toml [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
+y2 = [-16,-32,-48,-64,-80,-96,-112,-128,112,96,80,64,48,32,16,0]
 # all combinations of the high 2 bits + low 2 bits, i.e. xx0000xx in binary
-x3 = [0,-1,-2,-3,-64,-65,-66,-67,-128,-129,-130,-131,-192,-193,-194,-195]
-y3 = [0,-1,-2,-3,-64,-65,-66,-67,-128,-129,-130,-131,-192,-193,-194,-195]
+x3 = [-61,-62,-63,-64,-125,-126,-127,-128,67,66,65,64,3,2,1,0]
```
Comment (nickysn) [inline test_programs/plonky2_prove_failure/signed_lt_arr/Prover.toml:8]: Same remark as above - correct numbers, but in the wrong order.

File: test_programs/plonky2_prove_failure/signed_lt_arr/Prover.toml [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
+y3 = [-61,-62,-63,-64,-125,-126,-127,-128,67,66,65,64,3,2,1,0]
 # random numbers
-x4 = [0,-55,-64,-94,-98,-126,-128,-143,-161,-162,-163,-191,-199,-220,-229,-252]
-y4 = [0,-55,-64,-94,-98,-126,-128,-143,-161,-162,-163,-191,-199,-220,-229,-252]
+x4 = [-23,56,108,25,-123,-80,-90,-106,-107,45,-33,42,-37,-126,-6,101]
```
Comment (nickysn) [inline test_programs/plonky2_prove_failure/signed_lt_arr/Prover.toml:11]: Not in sorted order, either.

**Review by nickysn — CHANGES_REQUESTED** [review]
Please fix the order of the numbers in the test, otherwise it will fail when a correct signed integer comparison is implemented.

**Comment by Syndamia** [comment]
Issues are resolved in latest commit, thank you for the catch. I had sorted them in descending bit-wise order (which I just now realized why it was wrong).

**Review by nickysn — APPROVED** [review]
Looks good!

**Pull request merged**